File: doc-examples/example-kotlin/src/main/kotlin/example/StudentRepository.kt
package example

import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.Repository
import io.micronaut.data.annotation.Version
import io.micronaut.data.repository.CrudRepository

// tag::studentRepository[]
@Repository
interface StudentRepository : CrudRepository<Student, Long> {

    fun update(@Id id: Long, @Version version: Long, name: String)

    fun delete(@Id id: Long, @Version version: Long)

}
// end::studentRepository[]

File: doc-examples/hibernate-example-python/src/main/python/example/StudentRepository.py
# tag::studentRepository[]
from typing import Annotated

from micronaut.data.annotation import Id, Repository, Version
from micronaut.data.repository import CrudRepository

from example.Student import Student


@Repository
class StudentRepository(CrudRepository[Student, int]):

    def update(self, id: Annotated[int, Id], version: Annotated[int, Version], name: str) -> None: ...

    def delete(self, id: Annotated[int, Id], version: Annotated[int, Version]) -> None: ...
# end::studentRepository[]
